pynumaflow/reducer/_dtypes.py

```diff
@@ -232,4 +233,52 @@ def keys(self) -> list[str]:
         return self._key
 
 
-ReduceCallable = Callable[[list[str], AsyncIterable[Datum], Metadata], Awaitable[Messages]]
+ReduceAsyncCallable = Callable[[list[str], AsyncIterable[Datum], Metadata], Awaitable[Messages]]
+
+
+class Reducer(metaclass=ABCMeta):
+    """
+    Provides an interface to write a Reducer
+    which will be exposed over a gRPC server.
+    """
+
+    def __call__(self, *args, **kwargs):
+        """
+        Allow to call handler function directly if class instance is sent
+        as the reducer_instance.
+        """
+        return self.handler(*args, **kwargs)
+
+    @abstractmethod
+    async def handler(
+        self, keys: list[str], datums: AsyncIterable[Datum], md: Metadata
+    ) -> Messages:
+        """
+        Write a handler function which implements the ReduceCallable interface.
+        """
+        pass
+
+
+class ReduceBuilderClass:
```

Comment:
This is used just internally right? End user doesn't need to use this correct?

Reply:
Yes

Reply:
Then can you make it `_ReduceBuilderClass`?